protocol/messenger_contacts.go

```diff
@@ -1035,3 +1035,17 @@ func (m *Messenger) BuildContact(contactID string) (*Contact, error) {
 	}
 	return contact, nil
 }
+
+func (m *Messenger) BuildContactWithENSName(contactID, ENSName string) (*Contact, error) {
+	contact, ok := m.allContacts.Load(contactID)
+	if !ok {
+		var err error
+		contact, err = buildContactFromPkString(contactID)
+		contact.ENSVerified = true
```

Comment:
Sorry to ask, but I couldn't understand if/when we verified the ENS because here I only see the flag being set to true. It looks like `buildContactFromPkString` doesn't do any kind of ENS lookup. So what do we mean by `ENSVerified`?

Reply:
Thanks @cammellos. Now that you explained, I think it makes sense to not re-check. Normally, in centralized solutions where the backend is connected to any number of clients, it would normally re-validate anything done by clients, but since status-go + client is a completely different beast, it seems like re-checking would be a waste of resources.

Reply:
> In this case we don't need to re-resolve the ens, since the client has already verified it.

yep that's it